src/AspNetCore/Src/applicationmanager.cxx

```diff
@@ -196,14 +211,26 @@ APPLICATION_MANAGER::RecycleApplication(
 {
     HRESULT          hr = S_OK;
     APPLICATION_INFO_KEY  key;
+    APPLICATION_INFO* pApplicationInfo;
 
     hr = key.Initialize(pszApplicationId);
     if (FAILED(hr))
     {
         goto Finished;
     }
-    AcquireSRWLockExclusive(&m_srwLock);
-    m_pApplicationInfoHash->DeleteKey(&key);
+
+    AcquireSRWLockExclusive(&m_srwLock); 
+    m_pApplicationInfoHash->FindKey(&key, &pApplicationInfo); // FindKey will reference here.
+    ReleaseSRWLockExclusive(&m_srwLock);
```

Comment:
there could be a race condition that shutdown is called more than once. You can volatile flag to guard it

Reply:
Reference a local copy of the application info here to avoid this race.